### ISSUE TITLE
virtual_disk no datacenter_id and datastore_id

### ISSUE THREAD

**Issue by armand-s2h** [hashicorp/terraform-provider-vsphere#1456]
### Terraform Version
v1.0.0

### vSphere Provider Version
+ provider registry.terraform.io/hashicorp/vsphere v2.0.1

### Affected Resource(s)
- `vsphere_virtual_disk`

### Terraform Configuration Files

```hcl
resource "vsphere_virtual_disk" "data-disk" {
  size       = 6
  vmdk_path  = "data-disk.vmdk"
  datacenter = "Sandbox"
  #datastore  = "datastore_001"
  datastore     = "${data.vsphere_datastore.datastore.id}"
  type       = "thin"
}
```

### Debug Output

https://gist.github.com/armand-s2h/6a6f2534cb39a572e7973f843ec64248

### Panic Output

N/A

### Expected Behavior

`datastore_id` should work.

### Actual Behavior

With `datastore` can't find datastore with `datastore_id` : `Unsupported argument`
I have a similar problem with `datastore_id`. I have to hard code the values I can't use variables (example: `"${data.vsphere_datastore.datastore.id}"`)

### References

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by tenthirtyam** [comment]
@armand-s2h - The issue about is that you are passing the `.id` instead of `.name` for the `datastore`.

The following example works with the `latest` (v2.0.2) of the provider.

```hcl
terraform {
  required_providers {
    vsphere = {
      source  = "hashicorp/vsphere"
      version = ">= 2.0.2"
    }
  }
  required_version = ">= 1.0.10"
}

provider "vsphere" {
  user                 = "administrator@vsphere.local"
  password             = "VMware1!"
  vsphere_server       = "m01-vc01.rainpole.io"
  allow_unverified_ssl = true
}

data "vsphere_datacenter" "datacenter" {
  name = "m01-dc01"
}

data "vsphere_datastore" "datastore" {
  name          = "local-ssd-02"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

resource "vsphere_virtual_disk" "gh-1456" {
  size       = 100
  vmdk_path  = "gh-1456/foo.vmdk"
  datacenter = data.vsphere_datacenter.datacenter.name
  datastore  = data.vsphere_datastore.datastore.name
  type       = "eagerZeroedThick"
}
```

Results:

```bash
johnsonryan@johnsonryan-a03 ~/D/l/win-custom> terraform apply --auto-approve

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # vsphere_virtual_disk.gh-1456 will be created
  + resource "vsphere_virtual_disk" "gh-1456" {
      + adapter_type = "lsiLogic"
      + datacenter   = "m01-dc01"
      + datastore    = "local-ssd-02"
      + id           = (known after apply)
      + size         = 100
      + type         = "eagerZeroedThick"
      + vmdk_path    = "gh-1456/foo.vmdk"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
vsphere_virtual_disk.gh-1456: Creating...
vsphere_virtual_disk.gh-1456: Still creating... [10s elapsed]
vsphere_virtual_disk.gh-1456: Still creating... [20s elapsed]
vsphere_virtual_disk.gh-1456: Still creating... [30s elapsed]
vsphere_virtual_disk.gh-1456: Still creating... [40s elapsed]
vsphere_virtual_disk.gh-1456: Still creating... [50s elapsed]
vsphere_virtual_disk.gh-1456: Still creating... [1m0s elapsed]
vsphere_virtual_disk.gh-1456: Creation complete after 1m3s [id=gh-1456/foo.vmdk]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
In the vSphere (7.0 U3a) Client:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/7771363/140450405-dd073b15-5d01-4c02-9e6d-0bf4b8e3075b.png">

Hope this helps,
Ryan
@tenthirtyam

Recommend: `close/not-a-bug`

**Comment by tenthirtyam** [comment]
cc @iBrandyJackson and @appilon for review and closure.

The issue here based on the `vsphere_virtual_disk` resource follows the convention of using name .vs id. 

For example, `datacenter` is used vs. `datacenter_id`and `datastore` vs. `datastore_id` as seen in other resources (_e.g._, `vsphere_virtual_machine`). 

**TL;DR**, this is not a bug and is resolved by using `.name` from the datasource vs `.id`.

```hcl
data "vsphere_datacenter" "datacenter" {
  name = "m01-dc01"
}

data "vsphere_datastore" "datastore" {
  name          = "local-ssd-02"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

resource "vsphere_virtual_disk" "gh-1456" {
  size       = 100
  vmdk_path  = "gh-1456/foo.vmdk"
  datacenter = data.vsphere_datacenter.datacenter.name
  datastore  = data.vsphere_datastore.datastore.name
  type       = "eagerZeroedThick"
}
```

Ryan

**Comment by appilon** [comment]
Thanks for the breakdown @tenthirtyam. It's unfortunate if we created any accidental inconsistency between resources, but there could be technical reasons on why we reference by name and not id, in some cases I would have to dive into the resources to see if we could support both. Closing though since there is no bug.
